### DISCUSSION TITLE
How to pass arguments to node process during tests/benchmarks?

### DISCUSSION THREAD

**Discussion by robbiespeed** [vitest-dev/vitest#4954] [Q&A] [answered]
Particularly with mocha I can do `mocha --v8-expose-gc` and it will pass the argument to the underlying node process.

This is necessary for testing behavior that relies on WeakRef/FinalizationRegistry, and for testing against possible memory leaks.

**Comment by AriPerkkio** [comment, marked as answer]
There is [`poolOptions.<name>.execArgv` option](https://v3.vitest.dev/config/#pooloptions-threads-execargv). 

(Vitest v4: https://vitest.dev/config/execargv.html#execargv)

**Reply by acheronfail** [reply]
For anyone else needing it, this worked for me:

```ts
import type { UserConfig } from 'vite';
import { defineConfig } from 'vite';

// https://vitejs.dev/config
export default defineConfig(() => {
  return {
    test:
      poolOptions: {
        forks: {
          execArgv: ['--expose-gc'],
        },
      },
    },
  } as UserConfig;
});
```

**Reply by Vanilagy** [reply]
There isn't

**Reply by AriPerkkio** [reply]
It's just [top level `execArgs`](https://vitest.dev/config/execargv.html#execargv) since Vitest v4:  https://vitest.dev/guide/migration.html#pool-rework

```ts
export default defineConfig({
  test: {
    execArgv: ['--expose-gc'],
  }
})
```
